arrow-fx-reactor/src/main/kotlin/arrow/fx/reactor/extensions/ForwardDisposable.kt

```diff
@@ -0,0 +1,70 @@
+package arrow.fx.reactor.extensions
+
+import arrow.fx.internal.Platform
+import kotlinx.atomicfu.atomic
+import reactor.core.Disposable
+
+/**
+ * [ForwardDisposable] allows you to link 2 [Disposable]s.
+ *
+ * This is useful for when you want to link the cancel lifecycle of an `Emitter`
+ * to running an [Disposable] stream.
+ *
+ * This is used in `bracket`, `guaranteeCase` & `asyncF`.
+ */
+internal class ForwardDisposable {
```

Comment:
Nit: interface and create instances in the companion object

Reply:
Okay